libs/langchain/langchain/chat_models/azure_openai.py

```diff
@@ -51,48 +54,82 @@ class AzureChatOpenAI(ChatOpenAI):
     in, even if not explicitly saved on this class.
     """
 
-    deployment_name: str = Field(default="", alias="azure_deployment")
+    azure_endpoint: Union[str, None] = None
+    """Your Azure endpoint, including the resource.
+    
+        Example: `https://example-resource.azure.openai.com/`
+    """
+    deployment_name: Union[str, None] = Field(default=None, alias="azure_deployment")
+    """A model deployment. 
+    
+        If given sets the base client URL to include `/deployments/{azure_deployment}`.
+        Note: this means you won't be able to use non-deployment endpoints.
+    """
+    openai_api_version: str = Field(default="", alias="api_version")
+    """Automatically inferred from env var `OPENAI_API_VERSION` if not provided."""
+    openai_api_key: Union[str, None] = Field(default=None, alias="api_key")
+    """Automatically inferred from env var `AZURE_OPENAI_API_KEY` if not provided."""
+    azure_ad_token: Union[str, None] = None
+    """Your Azure Active Directory token.
+    
+        Automatically inferred from env var `AZURE_OPENAI_AD_TOKEN` if not provided.
+        
+        For more: 
+        https://www.microsoft.com/en-us/security/business/identity-access/microsoft-entra-id.
+    """  # noqa: E501
+    azure_ad_token_provider: Union[str, None] = None
+    """A function that returns an Azure Active Directory token.
+        
+        Will be invoked on every request.
+    """
     model_version: str = ""
+    """Legacy, for openai<1.0.0 support."""
     openai_api_type: str = ""
-    openai_api_base: str = Field(default="", alias="azure_endpoint")
-    openai_api_version: str = Field(default="", alias="api_version")
-    openai_api_key: str = Field(default="", alias="api_key")
-    openai_organization: str = Field(default="", alias="organization")
-    openai_proxy: str = ""
+    """Legacy, for openai<1.0.0 support."""
+    validate_base_url: bool = True
 
     @root_validator()
     def validate_environment(cls, values: Dict) -> Dict:
         """Validate that api key and python package exists in environment."""
-        values["openai_api_key"] = get_from_dict_or_env(
-            values,
-            "openai_api_key",
-            "OPENAI_API_KEY",
+        if values["n"] < 1:
```

Comment:
Maybe we document that this is coming from the parent class? I thought this was a bug when I saw it first